modules/sara_r4/cellular_r4_urc_handler.c

```diff
@@ -505,22 +505,30 @@ static void _cellular_UrcProcessUusord( CellularContext_t * pContext,
         /* Call the callback function of this session. */
         if( atCoreStatus == CELLULAR_AT_SUCCESS )
         {
-            pSocketData = _Cellular_GetSocketData( pContext, socketIndex );
-
-            if( pSocketData == NULL )
+            if( socketIndex == INVALID_SOCKET_INDEX )
             {
-                LogError( ( "_cellular_UrcProcessUusord : invalid socket index %d", socketIndex ) );
+                LogWarn( ( "_cellular_UrcProcessUusord : unknow session data received. "
+                           "The session %u may not be closed properly in previous execution.", sessionId ) );
             }
             else
             {
-                /* Indicate the upper layer about the data reception. */
-                if( pSocketData->dataReadyCallback != NULL )
+                pSocketData = _Cellular_GetSocketData( pContext, socketIndex );
+
+                if( pSocketData == NULL )
                 {
-                    pSocketData->dataReadyCallback( pSocketData, pSocketData->pDataReadyCallbackContext );
+                    LogError( ( "_cellular_UrcProcessUusord : invalid socket index %d", socketIndex ) );
                 }
                 else
                 {
-                    LogDebug( ( "_cellular_UrcProcessUusord: Data ready callback not set!!" ) );
+                    /* Indicate the upper layer about the data reception. */
+                    if( pSocketData->dataReadyCallback != NULL )
+                    {
+                        pSocketData->dataReadyCallback( pSocketData, pSocketData->pDataReadyCallbackContext );
+                    }
+                    else
+                    {
+                        LogDebug( ( "_cellular_UrcProcessUusord: Data ready callback not set!!" ) );
```

Comment:
Can it happen in a legitimate case? If not, it should probably be a warning or error. 

Can the application still function? If yes, it should be warning, else error.

Reply:
The application may use polling method to read socket data. It is legal not to register the callback function.

Reply:
Apologies but then you were right and it should be debug.